plasma/plasma-mvp/morevp-alice-bob.md

```diff
@@ -0,0 +1,103 @@
+MoreVP Alice-Bob Scenarios
+---
+
+Note that these scenarios work with N participants, not just limited to 1 input 1 output. 
+
+Also, note that in all of these scenarios the exits are given the priority of the youngest input to the transaction.
+
+#### Alice & Bob are honest and cooperating:
+
+Happy path.
+
+1. Alice spends UTXO1 in TX1 to Bob.
+2. Operator withholds, neither Alice nor Bob know if the transaction has been included. 
+3. Someone with access to TX1 (Alice, Bob, or otherwise) starts an exit and places `exit bond`.
+4. Alice and Bob "piggyback" onto the exit and each put up `piggyback bond`.
+5. After 1 week, Bob receives the money as if the transaction were included. All bonds are refunded.
+
+
+#### Bob spends:
+
+Bob trying to cheat.
+
+1. Alice spends UTXO1 in TX1 to Bob, creating UTXO2.
+2. TX1 is included in block N.
+3. Bob spends UTXO2 in TX2.
+4. TX2 is included in block N+M.
+5. Bob starts an exit from TX1.
+6. Bob "piggybacks" onto the exit and places `piggyback bond`
+7. Someone reveals TX2 spending UTXO2, so Bob's `piggyback bond` is slashed.
+
+
+#### Alice double spends:
+
+This is the case where Alice is malicious and has a double-spend secretly included before her transaction to Bob.
+
+1. Alice spends UTXO1 in TX1 to Bob.
+2. Operator withholds, neither Alice nor Bob know if the transaction has been included. 
+3. Unbeknownst to Bob, Alice has double spent UTXO1 in TX2. TX2 is included in the withheld block, but TX1 isn't.
```

Comment:
what if TX1 is included? how is this scenario then different?

Reply:
If `TX1` is included after `TX2`, then the scenario is exactly the same. If `TX1` is included before, then `TX1` is the canonical spend and `TX2` is the double spend.

Reply:
In the contract I represent this as:
not included => priority = MAX_UINT256
otherwise => prioirty = position in chain

So any in-flight transaction has lower priority than any included transaction, and then the first included spend of any of the transaction's inputs is the "canonical" spend.